[docs] add more examples on the spring store API docs (#7463)
sveltejs/svelte: site/content/docs/03-run-time.md

---

> The `store.stiffness`, `store.damping` and `store.precision` properties can be changed while the spring is in motion, and will take immediate effect.

Reviewer: Since this is directly after the bullet points, we could probably say this instead to not repeat ourselves.

```suggestion
All of the options above can be changed while the spring is in motion, and will take immediate effect.
```

Note: I use 'options' because the previous paragraph also refers them as "The following options are ..."

```js
const size = spring(100);
size.stiffness = 0.3;
size.damping = 0.4;
size.precision = 0.005;
```

---

As with [`tweened`](/docs#run-time-svelte-motion-tweened) stores, `set` and `update` return a Promise that resolves if the spring settles.

Both `set` and `update` can take a second argument — an object with `hard` or `soft` properties. `{ hard: true }` sets the target value immediately; `{ soft: n }` preserves existing momentum for `n` seconds before settling. `{ soft: true }` is equivalent to `{ soft: 0.5 }`.

```js
const coords = spring({ x: 50, y: 50 });
// updates the value immediately
coords.set({ x: 100, y: 200 }, { hard: true });
// preserves existing momentum for 1s

> coords.update((target_coords, coords) => { return { x: target_coords.x, y: coords.y }; }, { soft: 1 });

Reviewer: We might want to break this up into multiple lines so it's easier to read at a glance, this is how it looks from prettier
```suggestion
coords.update(
	(target_coords, coords) => {
		return { x: target_coords.x, y: coords.y };
	},
	{ soft: 1 }
);
```
